inspirehep/modules/orcid/api.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""APIs for ORCID."""
+
+from __future__ import absolute_import, division, print_function
+
+import requests
+from flask import current_app as app
+from inspire_utils.record import get_value
+from orcid import MemberAPI
+
+from inspirehep.modules.orcid import OrcidConverter
+from inspirehep.modules.orcid.utils import LOGGER, \
+    _get_api_url_for_recid, _split_lists, WORKS_BULK_QUERY_LIMIT, \
+    RECID_FROM_INSPIRE_URL
+
+
+def push_record_with_orcid(recid, orcid, oauth_token, put_code=None):
+    """Push record to ORCID with a specific ORCID ID.
+
+    Args:
+        recid (string): HEP record to push
+        orcid (string): ORCID identifier to push onto
+        oauth_token (string): ORCID user OAUTH token
+        put_code (Union[string, NoneType]): put-code to push record onto,
+            if None will push as a new record
+
+    Returns:
+        string: the put-code of the updated/inserted item
+    """
+    server_name = app.config["SERVER_NAME"]
+
+    record = _get_hep_record(app.config, recid)
+    if not record:
+        LOGGER.error('Cannot push record #{}, as metadata don\'t exist'.format(recid))
+
+    try:
+        bibtex = _get_bibtex_record(app.config, recid)
+    except requests.HTTPError:
+        bibtex = None
+        LOGGER.warning('Pushing record #{} without BibTex, as fetching'
+                       'it failed!'.format(recid))
+
+    orcid_api = _get_api()
+
+    orcid_xml = OrcidConverter(
+        record, server_name, put_code=put_code, bibtex_citation=bibtex
+    ).get_xml()
+
+    if put_code:
+        LOGGER.info("Pushing record #{} with put-code {} to ORCID {}.".format(
+            recid,
+            put_code,
+            orcid
+        ))
+        orcid_api.update_record(
+            orcid_id=orcid,
+            token=oauth_token,
+            request_type='work',
+            data=orcid_xml,
+            put_code=put_code,
+            content_type='application/orcid+xml',
+        )
+    else:
+        LOGGER.info(
```

Comment:
Why are we logging so much? Do we need it? Is this going o Sentry?
We probably only want to log in case of error.

Reply:
> Why are we logging so much?

I can't answer that one :)

> Do we need it?

I'd say it's useful to some extent (debugging, log archaeology after the fact...).

> Is this going o Sentry?

Nop, the info is not going to sentry, only the exception and error.